pkg/master/publish.go

```diff
@@ -128,15 +127,20 @@ func (m *Master) createMasterServiceIfNeeded(serviceName string, serviceIP net.I
 	return err
 }
 
-// setEndpoints sets the endpoints for the given service.
-// TODO: in a multi-master scenario this needs to consider all masters.
+// setEndpoints sets the endpoints for the given service. setEndpoints expects
+// that the endpoints objects it manages will all be managed only by
+// setEndpoints; therefore, to understand this, you need only understand the
+// requirements and the body of this function.
+//
+// Requirements:
+//  * All apiservers MUST use the same ports for their {rw, ro} services.
+//  * All apiservers MUST use setEndpoints and only setEndpoints to manage the
+//      endpoints for their {rw, ro} services.
+//  * All apiservers MUST know and agree on the number of apiservers expected
+//      to be running (m.masterCount).
+//  * setEndpoints is called periodically from all apiservers.
+//
```

Comment:
The first sentence of your comment ("setEndpoint sets...") makes this function sound like it is generic, while the Requirements section makes it clear that really this is only for registering the apiserver services. Can you add that caveat to the first paragraph as well, so people who aren't paying close attention don't get confused and think this should be used for other services?